### DISCUSSION TITLE
CNAME DNS records cannot be seen by dig or Resolve-DnsName

### DISCUSSION THREAD

**Discussion by momanten** [vercel/community#3703] [Help] [answered]
### Summary

Hello,
My domains CNAME DNS records cannot be seen in dns queries despite my domain is working and I see not any errors.
I would need CNAME records for my domain to use a Mongo App service feature.

### Steps to Reproduce

I have bought a domain for my hobby project (shibulingo.com) , created a CNAME record as it is needed for my Mongo DB App service for custom domain and validation. The CNAME record is there in the DNS records as I added and saved but after 4 days the CNAME records still cannot be seen by dns queries. 
 Resolve-DnsName -Name www.shibulingo.com -Type CNAME

Name                        Type TTL   Section    PrimaryServer               NameAdministrator           SerialNumber
----                        ---- ---   -------    -------------               -----------------           ------------
shibulingo.com              SOA  60    Authority  ns1.vercel-dns.com          hostmaster.nsone.net        1692267502

I only get back a SOA record. Why is it so ? Why neither the CNAME records cannot be seen in the CNAME record DNS queries, nor I can see any error at vercel. I would need it for my app to have a custom domain name other than the default mongodb one.

Thank you for the help in advance!

**Comment by amyegan** [comment, marked as answer]
Hey @momanten. It looks like [www.shibulingo.com](http://www.shibulingo.com/) redirects to [shibulingo.com](http://shibulingo.com/) as expected. Are you trying to use a different subdomain or [wildcard domain](https://vercel.com/docs/projects/domains/working-with-domains#wildcard-domain)?

**Reply by momanten** [reply]
Hi Amy! Thank you for your reply! No the problem is nto that. The domain is working however I need to have the custom domain feature from Mongo App Services. ( See here : https://www.mongodb.com/docs/atlas/app-services/hosting/use-a-custom-domain-name/#use-a-custom-domain-name ). For that I need to register 2 CNAME DNS records. First one for validation. I have already made this record for shibulingo.com. It is there , VErcel shows no errors that it would be problematic or anything, however the CNAME record queries gives back only a SOA record instead of the CNAME record. Mongo feature however searches for CNAME records for validation, so my domain name cannot be validated by them.

Result: 

dig  @ns1.vercel-dns.com. shibulingo.com CNAME

; <<>> DiG 9.10.6 <<>> @ns1.vercel-dns.com. shibulingo.com CNAME
; (2 servers found)
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 58094
;; flags: qr rd ra; QUERY: 1, ANSWER: 0, AUTHORITY: 1, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 1220
;; QUESTION SECTION:
;shibulingo.com.			IN	CNAME

;; AUTHORITY SECTION:
shibulingo.com.		60	IN	SOA	ns1.vercel-dns.com. hostmaster.nsone.net. 1692267502 43200 7200 1209600 60

or on windows 

Resolve-DnsName -Name [www.shibulingo.com](http://www.shibulingo.com/) -Type CNAME

Name Type TTL Section PrimaryServer NameAdministrator SerialNumber
shibulingo.com SOA 60 Authority ns1.vercel-dns.com hostmaster.nsone.net 1692267502

So I dont see my CNAME record. Why is it so why is it not working , not shown in the queries?


-----------
attached picture from vercel domain setting. it shows the cname record is there.

<img width="827" alt="Screenshot_vercel_cname" src="https://github.com/vercel/community/assets/33519468/dacf944e-a3bc-46b0-a781-5ab608fbd964">

Thank you for your help!, Br, Zoli

**Reply by amyegan** [reply]
When I looked for the CNAME record for that subdomain, I was able to find it. https://www.whatsmydns.net/#CNAME/_35125b768e584e5916b9a89ad4dbb45e.shibulingo.com

**Reply by momanten** [reply]
Hi!

Thank you Amy! I got answer even from Vercel Support. Thank you all! I have never worked with the CNAME records and thought simply querying the main domain should be enough to get the CNAME records as well as it works for A records... . Now tried as well and yes its working this way if I query with the full name not only the domain: 
  ------------ _35125b768e584e5916b9a89ad4dbb45e.shibulingo.com (23/08/2023 09:16:07) ------------

IP:


Detail:
; (1 server found)
;; Got answer:
;; ->>HEADER<<- opcode: Query, status: No Error, id: 54845
;; flags: qr aa rd; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 1
;; WARNING: recursion requested but not available

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; UDP: 1232; code: NoError
;; QUESTION SECTION:
_35125b768e584e5916b9a89ad4dbb45e.shibulingo.com. 	IN 	CNAME

;; ANSWER SECTION:
_35125b768e584e5916b9a89ad4dbb45e.shibulingo.com. 	60 	IN 	CNAME 	_406b4c5b5a8f57733d32d912f4892752.yfpsdsbdrq.acm-validations.aws.

;; Query time: 72 msec
;; SERVER: 198.51.44.13#53
;; WHEN: Wed Aug 23 07:16:07 Z 2023
;; MSG SIZE  rcvd: 155
I am contacting with Mongo Team and asking them why there the validation process is not working.
Thanks your help again! Have a nice day! Br, Zoltan
